zzzz_modular_occulus/code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -0,0 +1,539 @@
+						break
+				for (var/obj/machinery/door/airlock/B in get_step(newtarget.loc, turn(direction, 90)))
+					if (istype(B, newtarget.type))
+						if (!check_target(B, user)) return
+						target_node2 = B
+						target_node2.bracer = src
+						break
+
+		user.drop_from_inventory(src, src.loc)
+
+		forceMove(get_step(newtarget.loc, reverse_direction(direction)))
+		set_dir(reverse_direction(direction))
+		status = STATUS_ACTIVE
+		attach(newtarget)
+		user.visible_message("<span class='notice'>[user] attached [src] onto [newtarget] and flicks it on. The magnetic lock now seals [newtarget].</span>", "<span class='notice'>You attached [src] onto [newtarget] and switched on the magnetic lock.</span>")
+		return
+
+
+/obj/item/device/magnetic_lock/proc/setconstructionstate(var/newstate)
+	if (!powercell && newstate == 1)
+		setconstructionstate(2)
+		return
+	else if (newstate == 4)
+		detach(playflick = 0)
+		update_icon()
+	constructionstate = newstate
+	update_icon()
+
+/obj/item/device/magnetic_lock/proc/detach(var/playflick = 1)
+	if (target)
+
+		if (playflick)
+			spawn(-15) flick("release_[department]", src)
+
+		status = STATUS_INACTIVE
+		set_dir(SOUTH)
+		update_icon()
+		layer = LAYER_NORMAL
+
+		target.bracer = null
+		target = null
+		if (target_node1)
+			target_node1.bracer = null
+			target_node1 = null
+		if (target_node2)
+			target_node2.bracer = null
+			target_node2 = null
+		anchored = 0
+
+		STOP_PROCESSING(SSobj, src)
+		last_process_time = 0
+
+/obj/item/device/magnetic_lock/proc/attach(var/obj/machinery/door/airlock/newtarget as obj)
+	layer = LAYER_ATTACHED
+
+	newtarget.bracer = src
+	target = newtarget
+
+	last_process_time = world.time
+	START_PROCESSING(SSobj, src)
+	anchored = 1
+
+	spawn(-15)
+		flick("deploy_[department]", src)
+	update_icon()
+
+/obj/item/device/magnetic_lock/update_icon()
+	if (status == STATUS_ACTIVE && target)
+		icon_state = "active_[department]"
+		switch (dir)
+			if (NORTH)
+				pixel_x = 0
+				pixel_y = -32
+			if (EAST)
+				pixel_x = -32
+				pixel_y = 0
+			if (SOUTH)
+				pixel_x = 0
+				pixel_y = 32
+			if (WEST)
+				pixel_x = 32
+				pixel_y = 0
+	else if (status >= STATUS_INACTIVE)
+		icon_state = "inactive_[department]"
+		pixel_x = 0
+		pixel_y = 0
+	else
+		icon_state = "broken_[department]"
+		pixel_x = 0
+		pixel_y = 0
+	update_overlay()
+
+/obj/item/device/magnetic_lock/proc/update_overlay()
+	cut_overlays()
+	switch (status)
+		if (STATUS_BROKEN)
+			icon_state = "broken"
+			return
+
+		if (STATUS_INACTIVE to STATUS_ACTIVE)
+			if (hacked)
+				add_overlay("overlay_hacked")
+			else if (locked)
+				add_overlay("overlay_locked")
+			else
+				add_overlay("overlay_unlocked")
+			switch (constructionstate)
+				if (0)
+					return
+				if (1 to 4)
+					add_overlay("overlay_deconstruct_[constructionstate]")
+
+/obj/item/device/magnetic_lock/proc/take_damage(var/damage)
+	if(invincible)
+		return
+	health -= rand(damage/2, damage)
+
+	if (damage >= 40 && prob(50))
```

Comment:
I thought this was over 20 damage? Not 40?

Reply:
This is a condition that says, if an obj's damage is over 40, it has a 50% chance to outright destroy the lock. A la, Antimaterial rifle shots. Or super swords.